metis-enrichment/metis-enrichment-service/src/test/java/eu/europeana/enrichment/service/wikidata/WikidataAccessDaoTest.java

```diff
@@ -41,41 +47,129 @@ public void tearDown() throws Exception {}
+
+    //verify correct conversion to Core Organization
+    //in the future more labels might be available, therefore use in comparison and not equality
+    assertTrue(6 <= organizationImpl.getPrefLabel().values().size());
+    assertEquals(WIKIDATA_URL_NISV, organizationImpl.getAbout());
+    assertTrue(organizationImpl.getEdmAcronym().get("nl").contains(acronym));
+    //disabled address until the locality issue is solved
+    //assertNotNull(organizationImpl.getAddress());
+    //assertNotNull(organizationImpl.getAddress().getVcardStreetAddress()); 
+    //the locality was disabled in the current version of the mapping, but it might come back in the future 
+    //    assertNotNull(organizationImpl.getAddress().getVcardLocality());
+    //assertNotNull(organizationImpl.getAddress().getVcardCountryName());
+  }
+
+  private Organization convertToCoreOrganization(WikidataOrganization wikidataOrganization) {
+    //convert Wikidata organization to OrganizationImpl object
+    Organization organizationImpl = wikidataAccessService.toOrganizationImpl(wikidataOrganization);
+    assertNotNull(organizationImpl);
+    return organizationImpl;
+  }
 
+  private WikidataOrganization dereferenceWikidataOrg(String acronym, String uri)
+      throws WikidataAccessException, IOException, URISyntaxException, FileNotFoundException,
+      JAXBException {
     //dereference Wikidata URI
     String wikidataXml = wikidataAccessDao.getEntity(uri).toString();
     assertNotNull(wikidataXml);
 
     //write XML to a file (if it fails, it throws an exception). 
-    File wikidataOutputFile = getClasspathFile(WIKIDATA_TEST_OUTPUT_FILE);
+    File wikidataOutputFile = getDerefFile(acronym);
     wikidataAccessService.saveXmlToFile(wikidataXml, wikidataOutputFile);
 
-    //insert XML in Wikidata organization object 
-    WikidataOrganization wikidataOrganization = wikidataAccessDao.parse(wikidataXml);
-    assertNotNull(wikidataOrganization);
-    assertEquals(wikidataOrganization.getOrganization().getAbout(), TEST_WIKIDATA_URL);
-    assertEquals(wikidataOrganization.getOrganization().getCountry(), TEST_COUNTRY);
-
     //read organization XML from file
-    File wikidataTestOutputFile = getClasspathFile(WIKIDATA_TEST_OUTPUT_FILE);
-    String savedWikidataXml = FileUtils.readFileToString(wikidataTestOutputFile, "UTF-8"); 
-    assertEquals(wikidataXml, savedWikidataXml);
-
-    //convert Wikidata organization to OrganizationImpl object
-    Organization organizationImpl = wikidataAccessService.toOrganizationImpl(wikidataOrganization);
-    assertNotNull(organizationImpl);
-    assertEquals(38, organizationImpl.getPrefLabel().values().size());
-    assertEquals(organizationImpl.getAbout(), TEST_WIKIDATA_URL);
-    assertEquals(TEST_ACRONYM, organizationImpl.getEdmAcronym().get(Locale.FRENCH.getLanguage()).get(0));
+    //File wikidataTestOutputFile = getClasspathFile(WIKIDATA_TEST_OUTPUT_FILE);
```

Comment:
Why is this line commented out? Can we remove it?

Reply:
yes ..  I'll remove it

Reply:
done